### ISSUE TITLE
Make is_local recognize loopback addresses 127.0.1.1

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#23504]
127.0.1.1 is used in ubuntu / debian by default in /etc/hosts for fqdn.

Due to the fact that address don't literally match the address of network interface isLocal fails to detect that address as local.

See
https://www.debian.org/doc/manuals/debian-reference/ch05.en.html#_the_hostname_resolution
https://serverfault.com/questions/363095/why-does-my-hostname-appear-with-the-address-127-0-1-1-rather-than-127-0-0-1-in

https://github.com/ClickHouse/ClickHouse/blob/51ba12c2c3106d9277bbc223adfe41fdb3e45439/src/Common/isLocalAddress.cpp#L13

**Comment by filimonov** [comment]
Seem related to https://github.com/ClickHouse/ClickHouse/issues/9101

**Comment by alexey-milovidov** [comment]
Why it's needed in production use cases?

We use 127.0.0.{2..} in tests to distinguish these addresses from 127.0.0.1 and to check more cases.
So, if we will treat these addresses as local, our tests will cover less code paths.

**Comment by filimonov** [comment]
> Why it's needed in production use cases?

Reason is very simple. When you configure fqdn during ubuntu / debian installation (debian-installer-launcher)it adds a line 
```
127.0.1.1 myclickhouse1.host.name.com
```
to /etc/hosts (motivation is explained in https://serverfault.com/questions/363095/why-does-my-hostname-appear-with-the-address-127-0-1-1-rather-than-127-0-0-1-in )

and later when you use `myclickhouse1.host.name.com` in `<remote_servers>` section, it is resolved to `127.0.1.1`, and not recognized by clickhouse as local address. 

Since that configuration is default for ubuntu / debinal, i think clickhouse should be ready for that.

> We use 127.0.0.{2..} in tests to distinguish these addresses from 127.0.0.1 and to check more cases.
> So, if we will treat these addresses as local, our tests will cover less code paths.

I guess we can add a setting for tests, something like 
```
<consider_all_loopback_addresses_local>0</consider_all_loopback_addresses_local>
```

**Comment by alexey-milovidov** [comment]
> I guess we can add a setting

Adding a setting is too much.

**Comment by filimonov** [comment]
> > I guess we can add a setting
> 
> Adding a setting is too much.

Any better options if we want to have one behaviour for production usage, and another for tests ?  (the setting can also live in remote_servers).

**Comment by alexey-milovidov** [comment]
@filimonov ClickHouse works correctly regardless.
We can simply avoid implementing this feature.

And you will not often use the generated FQDN in configuration of your cluster.

For example, I have `127.0.1.1 github-explorer.ru-central1.internal github-explorer` on one of the machines, but the .internal address has no use in ClickHouse cluster configuration.

**Comment by filimonov** [comment]
On cluster DDL have issues with recognizing itself...

**Comment by alexey-milovidov** [comment]
It's better to avoid implementation of this feature.

**Comment by alexey-milovidov** [comment]
Unless you will show that there is popular scenario where this feature will make significant benefits.

**Comment by filimonov** [comment]
Well the problem is that for users which made a 'canonical setup' of Ubuntu / debian clickhouse can't work properly. And debugging the problem is not straight-forward. 

We should either support that default configuration, either warn user (for example during server start) that we don't support that. Second option is bit worse but acceptable.

**Comment by alexey-milovidov** [comment]
> made a 'canonical setup' of Ubuntu / debian clickhouse can't work properly

What is the issue? What is working in a wrong way?

**Comment by alexey-milovidov** [comment]
Suppose the host is named `myhost` and it is resolved to 127.0.1.1.
Why exactly we need to treat `myhost` as local? What is the scenario?
If we don't treat it as local, I assume that everything will work correctly.

**Comment by alexey-milovidov** [comment]
> and later when you use myclickhouse1.host.name.com in <remote_servers> section, it is resolved to 127.0.1.1, and not recognized by clickhouse as local address.

But everything should will work correctly regardless if we apply or don't apply localhost optimization.

**Comment by alexey-milovidov** [comment]
> on cluster DDL have issues with recognizing itself and queries hang. See #18341

We should rewrite the logic of ON CLUSTER queries to use host UUIDs instead of IP addresses.

**Comment by filimonov** [comment]
I said before - there are 2 main issues.
1) distributed ddl not working (it sees the task for `myclickhouse1.host.name.com` but it's resolved to 127.0.1.1 and which is 'not local') 
2) prefer localhost replica for distributed tables don't work (it thinks all nodes remote) 

Yes, alternative are also possible - just get rid of isLocalAddress in most of contextes.
